### DISCUSSION TITLE
How to add records to the file type field in a collection when sending requests using RESTful API

### DISCUSSION THREAD

**Discussion by dong-lufei** [pocketbase/pocketbase#3056] [Q&A]
What kind of request？
uploadFile specifies the temporary path to the file

**Comment by ganigeorgiev** [comment]
File upload is supported only when sending `multipart/form-data` request.

For an example you can check https://pocketbase.io/docs/files-handling/#uploading-files.

**Comment by dong-lufei** [comment]
@ganigeorgiev It was uploaded using pocketbase, instead of http, or curl

**Reply by ganigeorgiev** [reply]
I don't understand, do you mean when using PocketBase as framework?

In that case, you can check the `form.AddFiles()` example in:

- https://pocketbase.io/docs/go-records/#create-new-record-with-data-validations
- https://pocketbase.io/docs/js-records/#create-new-record-with-data-validations

_(the above examples are for create, but the `RecordUpsert` form operations are the same for both create and update)_

**Comment by dong-lufei** [comment]
My problem is that my client selects a temporary path for an image to get to it, and to make this temporary path permanent, I need to request modification of a file field for a record in a collection

My client environment can't use pocketbase, because npm i pocketbase, some of the methods in it won't work, there is no browser API for that environment

So I CURD: PATCH something like this
/api/collections/xxx/records/:id

**Reply by ganigeorgiev** [reply]
I'm failing to understand what do you mean. Please elaborate or provide a code sample (_pseudo-code is always fine_) with the desired behavior.

**Comment by dong-lufei** [comment]
```
const uploadTask = uni.uploadFile({
	url: "https://xxyy.com/release/api/collections/lala/records/mrc1jv28o3a92ng
", //仅为示例，非真实的接口地址
	filePath: avatarUrl, // AvatarUrl is the temporary path to the file
	name: "myAvatarUrl",
	success: (uploadFileRes) => {
		console.log("uploadTask-success", uploadFileRes.data)
	},
	fail: (err) => {
		console.log("uploadTask-fail", err)
	},
})
console.log("uploadTask", uploadTask)
```

**Reply by ganigeorgiev** [reply]
I don't know what `uni` is in the above.

In any case you can't submit a local file path as `file` field value because the PocketBase server will not have access to it. You need somehow to read the file and submit its content as binary blob. Maybe `uni.uploadFile` is doing this?

Additionally, when submitting files the content type of the request must be `multipart/form-data`.
I don't know how a `multipart/form-data` request can be submitted with the library you use. You have to research that on your own.

**Comment by dong-lufei** [comment]
```
const createdRecord = await pb
     .collection("lala")
     .update("mrc1jv28o3a92ng", formData)
console.log("🚀 ~ file: t.html:34 ~ createdRecord:", createdRecord)
```
This is what happens when you replace it with something like fetch(). Oh, form-data only. 555

**Comment by dong-lufei** [comment]
@ganigeorgiev I found a solution. It's a long way. My client sends a post request to upload a file to my proxy back-end, and my proxy back-end converts the received request body data into formdata and requests the pocketbase database address for storage by patch request
